### PR TITLE
Update changes for 0.6.0

### PR DESCRIPTION
**Pull request by chriseth**

### CODE REVIEW

File: docs/060-breaking-changes.rst [hunk 1]
```diff
@@ -102,27 +121,11 @@ This section gives detailed instructions on how to update prior code for every b
...
-
-Deprecated Elements
-===================
-
-This section lists changes that deprecate prior features or syntax.
-
-
-.. _interoperability_060:
-
-Interoperability With Older Contracts
```
Comment (chriseth) [inline docs/060-breaking-changes.rst:126]: We had that for 0.5.0, but I'm not sure if we really need it.

**Comment by chriseth** [comment]
We should probably also list changes outside of the language:
 - yul optimizer activated together with the legacy optimizer
 - abiv2 not considered experimental
 - c api changes
 - storage layout export
 - metadata hash options
 - new error reporter

File: docs/060-breaking-changes.rst [hunk 2]
```diff
@@ -9,28 +9,42 @@ For the full list check
...
-* Function ``push(value)`` for dynamic storage arrays do not return the new length anymore.
 
-* The new keyword ``abstract`` can be used to mark contracts as abstract. It has to be used
-  if a contract does not implement all its functions.
+Explicitness Requirements
+=========================
 
-* Libraries have to implement all their functions, not only the internal ones.
+This section lists changes where the code now needs to be more explicit,
+but the semantics to do not change.
```
Comment (bshastry) [inline docs/060-breaking-changes.rst]:
```suggestion
but the semantics do not change.
```

File: docs/060-breaking-changes.rst [hunk 3]
```diff
@@ -9,28 +9,42 @@ For the full list check
...
+  implementation outside an interface have to be marked ``virtual``.
+  When overriding a function or modifier, the new keyword ``override``
+  must be used. When overriding a function or modifier defined in multiple
+  parallel bases, all bases must be listed in parentheses after the keyword
+  like so: ``override(Base1, Base2)``.
 
-* Member-access to ``length`` of arrays is now always read-only, even for storage arrays. It's no
+* Member-access to ``length`` of arrays is now always read-only, even for storage arrays. It is no
   longer possible to resize storage arrays assigning a new value to their length. Use ``push()``,
   ``push(value)`` or ``pop()`` instead, or assign a full array, which will of course overwrite existing content.
```
Comment (bshastry) [inline docs/060-breaking-changes.rst:40]: Unrelated to this PR: Are the following examples equivalent?

Push:
```
contract C { bool[] array; function foo() public { array.push(true); }
```

Indexing:
```
contract C { bool[] array; function foo() public { array[0] = true; }
```

File: docs/060-breaking-changes.rst [hunk 4]
```diff
@@ -72,13 +81,21 @@ This section highlights changes that affect syntax and semantics.
...
-  When overriding a function or modifier, the new keyword ``override``
-  must be used. When overriding a function or modifier defined in multiple
-  parallel bases, all bases must be listed in parentheses after the keyword
-  like so: ``override(Base1, Base2)``.
+
+New Features
+============
+
+This section lists things that were not possible prior to Solidity 0.6.0
+or at least more difficult.
```
Comment (bshastry) [inline docs/060-breaking-changes.rst]:
```suggestion
or at least were more difficult to achieve prior to Solidity 0.6.0.
```

File: docs/060-breaking-changes.rst [hunk 5]
```diff
@@ -72,13 +81,21 @@ This section highlights changes that affect syntax and semantics.
...
+
+ * The :ref:`try/catch statement <try-catch>` allows you to react on failed external calls.
+ * ``struct`` and ``enum`` types can be declared at file level.
+ * Array slices can be used for calldata arrays, for example ``abi.decode(msg.data[4:], (uint, uint))``
+   is a low-level way to decode the function call payload.
+ * Natspec supports multiple return parameters in dev documentation, enforcing the same naming check as ``@param``.
```
Comment (bshastry) [inline docs/060-breaking-changes.rst]:
```suggestion
 * Natspec supports multiple return parameters in developer documentation, enforcing the same naming check as ``@param``.
```

**Review by bshastry — CHANGES_REQUESTED** [review]
Added minor comments, looks good to me in general.

File: docs/060-breaking-changes.rst (hunk 3 above)
Comment (leonardoalt) [reply docs/060-breaking-changes.rst:40]: The second will revert because `array.length = 0`.

**Review by bshastry — APPROVED** [review]
LGTM.

**Comment by bshastry** [comment]
> We should probably also list changes outside of the language:
> 
> * yul optimizer activated together with the legacy optimizer
> * abiv2 not considered experimental
> * c api changes
> * storage layout export
> * metadata hash options
> * new error reporter

Do you want to do these in this PR or a new one?

**Comment by erak** [comment]
@bshastry @chriseth Merging this now and created https://github.com/ethereum/solidity/issues/8007 to track the documentation of changes outside of the language.

**Pull request merged**